app/lua/lua.c

```diff
@@ -468,11 +324,8 @@ int lua_main (int argc, char **argv) {
 
 void lua_handle_input (bool force)
 {
-  while (gLoad.L && (force || readline (&gLoad)))
-  {
+  if (gLoad.L && (force || readline (&gLoad)))
```

Comment:
Whoa, this change makes me nervous. I explicitly changed it to a `while` to fix the issue of lost commands on the serial. If more than one line buffers up the LVM would fall behind in its input processing.

Reply:
To be honest, I can't remember why I did this.  I will take a look at the history.  My first pass through merged in the changes from a vanilla Lua-5.1 variant and I may just have missed this in the merge.  

But the whole interactive interface is very flaky anyway.  It is really designed for a more POSIX-like environment.  The way it determines compilation units is to stack the lies and try:
-  line 1
- line 1 + line 2
- line 1 + line 2 + line 3
- ...

Until the sequence compiles.  It then executes the brick and treats the next line as line 1.  In the meantime there is no flow control (hardware or XON/XOFF) on the input uart so it is _very_ easy to overflow the 128 byte input FIFO on the UART, with or without this change.

Also  the whole of the interactive interface including the input / output redirector is extremely flaky.   

However all of this is separate to this change so I will back this one out here.